### PR TITLE
YD-683 Don't fail pin reset when user is removed

### PR DESCRIPTION
**Pull request by Bert-R**
If a user decides to remove or overwrite their account while waiting for a pin reset confirmation code, the batch job generating the reset code should not fail. This is now addressed.

Along with this:
* Removed unused attribute messageSupplierCaptor from BuddyDeviceChangeMessageDtoTest

### CODE REVIEW

File: core/src/main/java/nu/yona/server/subscriptions/service/BuddyDeviceChangeMessageDto.java
Comment (yonabuild) [inline core/src/main/java/nu/yona/server/subscriptions/service/BuddyDeviceChangeMessageDto.java:113]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS00100)

File: core/src/main/java/nu/yona/server/subscriptions/service/BuddyDisconnectMessageDto.java
Comment (yonabuild) [inline core/src/main/java/nu/yona/server/subscriptions/service/BuddyDisconnectMessageDto.java:139]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS00100)

File: core/src/main/java/nu/yona/server/messaging/service/MessageDto.java
Comment (yonabuild) [inline core/src/main/java/nu/yona/server/messaging/service/MessageDto.java:193]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS00100)

File: core/src/main/java/nu/yona/server/subscriptions/service/BuddyConnectResponseMessageDto.java
Comment (yonabuild) [inline core/src/main/java/nu/yona/server/subscriptions/service/BuddyConnectResponseMessageDto.java:161]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS00100)

File: core/src/main/java/nu/yona/server/subscriptions/service/UserUpdateService.java
Comment (yonabuild) [inline core/src/main/java/nu/yona/server/subscriptions/service/UserUpdateService.java:344]: MAJOR A "NullPointerException" could be thrown; "buddy" is nullable here. [rule](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS2259)

File: core/src/main/java/nu/yona/server/subscriptions/service/UserUpdateService.java
Comment (yonabuild) [inline core/src/main/java/nu/yona/server/subscriptions/service/UserUpdateService.java:346]: MAJOR A "NullPointerException" could be thrown; "userEntity" is nullable here. [rule](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS2259)

File: core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java
Comment (yonabuild) [inline core/src/main/java/nu/yona/server/subscriptions/service/BuddyService.java:439]: MAJOR This method has 21 lines, which is greater than the 20 lines authorized. Split it into smaller methods. [rule](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS138)

File: core/src/main/java/nu/yona/server/subscriptions/service/BuddyInfoChangeMessageDto.java
Comment (yonabuild) [inline core/src/main/java/nu/yona/server/subscriptions/service/BuddyInfoChangeMessageDto.java:113]: MINOR Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS00100)

**Comment by yonabuild** [comment]
SonarQube analysis reported 8 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: core/src/main/java/nu/yona/server/subscriptions/service/UserUpdateService.java
Comment (Bert-R) [reply core/src/main/java/nu/yona/server/subscriptions/service/UserUpdateService.java:344]: Cannot be null. This is asserted on line 339.

File: core/src/main/java/nu/yona/server/subscriptions/service/UserUpdateService.java
Comment (Bert-R) [reply core/src/main/java/nu/yona/server/subscriptions/service/UserUpdateService.java:346]: Cannot be null. This is asserted on line 338.

File: core/src/main/java/nu/yona/server/subscriptions/service/UserService.java
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/subscriptions/service/UserService.java:323]: How does this work when `lockUserForUpdate` is called and an exception occurs? Is the lock released because the transaction failed?

**Review by aristotelos — COMMENTED** [review]
Reviewed, because it is so large I was unable to look at all the details. In many places there is a change from DTO to entity in the code, which I think is not bad, but the reason is not entirely clear to me. I also see that locking of the user entity for update is added in multiple places. Is that only for the pin reset or is it a general improvement on race conditions?

Moreover, getting to the reason of the PR, I see no changes in `sendPinResetConfirmationCode` except that it locks the user for update. So that will throw if the user is removed, while the previous `getUserEntityById` would not throw?

File: core/src/main/java/nu/yona/server/subscriptions/service/PinResetRequestService.java
Comment (aristotelos) [inline core/src/main/java/nu/yona/server/subscriptions/service/PinResetRequestService.java:74]: I don't get it. Does this just silently proceed when the user is removed?

**Comment by Bert-R** [comment]
> Reviewed, because it is so large I was unable to look at all the details. In many places there is a change from DTO to entity in the code, which I think is not bad, but the reason is not entirely clear to me. I also see that locking of the user entity for update is added in multiple places. Is that only for the pin reset or is it a general improvement on race conditions?
> 
> Moreover, getting to the reason of the PR, I see no changes in `sendPinResetConfirmationCode` except that it locks the user for update. So that will throw if the user is removed, while the previous `getUserEntityById` would not throw?

Note that after the initial commit and PR creation, fixes for several related issues were pushed to this PR. The first issue, related to a PIN reset for a removed user is resolved in the first commit (7ac06bc69ab2236d609e9d4a8ddf3a2af6fb11fb) at line 69 of PinResetRequestService.java

File: core/src/main/java/nu/yona/server/subscriptions/service/UserService.java
Comment (Bert-R) [reply core/src/main/java/nu/yona/server/subscriptions/service/UserService.java:323]: Indeed. Once the transaction completes (whether successfully or in error), the lock is released.

File: core/src/main/java/nu/yona/server/subscriptions/service/PinResetRequestService.java
Comment (Bert-R) [reply core/src/main/java/nu/yona/server/subscriptions/service/PinResetRequestService.java:74]: See my other comment.

**Comment by Bert-R** [comment]
There's one more issue with this PR, causing the ``ActivityTest`` integration test to fail. So don't merge this PR yet.

**Comment by Bert-R** [comment]
This one I take forward once #592 is merged. The issue is that originally, we initialized the ``GoalIdMapping`` using the DTOs and now using the entities. The DTOs have a collection ``goals`` which actually contains ``goalsIncludingHistoryItems``. Because of that, the tests that use activities on updated goals now fail, as the history items are not available in the mapping. Once the mentioned PR is merged, I'm going to see what's the best way to fix this.

**Comment by aristotelos** [comment]
👍 #592 is now merged. 

> You're either extremely smart or extremely lucky.

You already should know me by now, right? 🤣

**Comment by Bert-R** [comment]
With the last update all tests work and the change in entity loading is reasonable. I consider this PR ready for merging.

File: core/src/main/java/nu/yona/server/subscriptions/service/GoalIdMapping.java
Comment (yonabuild) [inline core/src/main/java/nu/yona/server/subscriptions/service/GoalIdMapping.java:13]: MINOR Remove this unused import 'nu.yona.server.goals.entities.Goal'. [rule](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by yonabuild** [comment]
SonarQube analysis reported 9 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 6 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by yonabuild** [comment]
SonarQube analysis reported 8 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by aristotelos — APPROVED** [review]

**Pull request merged**
